src/gtkspellcheck/spellcheck.py

```diff
@@ -663,43 +723,41 @@ def _check_word(self, start, end):
         for tag in self.ignored_tags:
             if start.has_tag(tag):
                 return
-        if _py3k:
-            word = self._buffer.get_text(start, end, False).strip()
-        else:
-            word = self._buffer.get_text(start, end, False).decode("utf-8").strip()
+        word = self._buffer.get_text(start, end, False).strip()
         if not word:
             return
         if len(self._filters[SpellChecker.FILTER_WORD]):
             if self._regexes[SpellChecker.FILTER_WORD].match(word):
                 return
         if len(self._filters[SpellChecker.FILTER_LINE]):
-            line_start = self._buffer.get_iter_at_line(start.get_line())
+            if Gtk.MAJOR_VERSION > 3:
+                _success, line_start = self._buffer.get_iter_at_line(start.get_line())
+            else:
+                line_start = self._buffer.get_iter_at_line(start.get_line())
             line_end = end.copy()
             line_end.forward_to_line_end()
-            if _py3k:
-                line = self._buffer.get_text(line_start, line_end, False)
-            else:
-                line = self._buffer.get_text(line_start, line_end, False).decode(
-                    "utf-8"
-                )
+            line = self._buffer.get_text(line_start, line_end, False)
             for match in self._regexes[SpellChecker.FILTER_LINE].finditer(line):
                 if match.start() <= start.get_line_offset() <= match.end():
-                    start = self._buffer.get_iter_at_line_offset(
-                        start.get_line(), match.start()
-                    )
-                    end = self._buffer.get_iter_at_line_offset(
-                        start.get_line(), match.end()
-                    )
+                    if Gtk.MAJOR_VERSION > 3:
+                        _success, start = self._buffer.get_iter_at_line_offset(
```

Comment:
Maybe add a comment explaining that this should always succeed as the positions come from the buffer itself?

Reply:
Done